File: leetcode/go/2239.find-closest-number-to-zero.go
func findClosestNumber(nums []int) int {
    ans := 10000000
    dist := ans
    for _, n := range nums {
        y := Abs(n)
        if y < dist {
            ans = n
            dist = y
        } else if y == dist {
            ans = Max(ans, n)
        }
    }
    return ans
}

func Abs(x int) int {
    if x < 0 {return -x}
    return x
}

func Max(x int, y int) int {
    if x > y {return x}
    return y
}

File: leetcode/rust-leetcode/src/2239.find-closest-number-to-zero.rs
impl Solution {
    pub fn find_closest_number(nums: Vec<i32>) -> i32 {
        let mut ans = i32::MAX;
        let mut dist = i32::MAX;
        for &n in nums.iter() {
            if n.abs() < dist {
                ans = n;
                dist = n.abs();
            } else if n.abs() == dist {
                ans = ans.max(n);
            }
        }
        ans
    }
}

